enhancements/subscription-content/subscription-injection.md

```diff
@@ -0,0 +1,487 @@
+So that I download RHEL content from my Satellite instances instead of the Red Hat CDN.
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Installation and Setup
+
+When the `subscription-injection-operator` is installed, it creates the subscription `Bundle` and
+`ClusterBundle` custom resource definition. The operator deployment creates sample YAML files
+which helps admins get started with default, cluster-wide and namespace-scoped subscription
+bundles:
+
+```yaml
+apiVersion: console.openshift.io/v1
+kind: ConsoleYAMLSample
+metadata:
+  name: subscription-cluster-bundle-example
+spec:
+  targetResource:
+    apiVersion: subscription.openshift.io/v1
+    kind: ClusterBundle
+  title: Cluster-wide subscription bundle example
+  description: | 
+    An example of a subscription bundle that can be injected into any workload that has the
+    `subscription.openshift.io/inject-cluster-bundle: cluster` annotation in its pod spec, and
+    whose service account has the `edit` or `admin` role.
+  yaml: |
+    apiVersion: subscription.openshift.io/v1
+    kind: ClusterBundle
+    metadata:
+      name: cluster
+    spec:
+      aggregateToClusterRoles:
+      - edit
+      - admin
+      entitlements:
+      - name: etc-pki-entitlement
+        namespace: openshift-config
+```
+
+```yaml
+apiVersion: console.openshift.io/v1
+kind: ConsoleYAMLSample
+metadata:
+  name: subscription-namespaced-bundle-example
+spec:
+  title: Namespaced subscription bundle example
+  description: |
+    An example of a subscription bundle that can be injected into any workload in the bundle's
+    namespace that has the `subscription.openshift.io/inject-bundle: subscription` annotation in
+    its pod spec.
+  targetResource:
+    apiVersion: subscription.openshift.io/v1
+    kind: Bundle
+  yaml: |
+    apiVersion: subscription.openshift.io/v1
+    kind: Bundle
+    metadata:
+      name: subscription
+    spec:
+      entitlements:
+      - name: etc-pki-entitlement
+```
+
+The `entitlements` object is a reference to the `Secrets` which contain the entitlement keys for
+the cluster's subscription. The cluster administrator is responsible for obtaining the entitlement
+keys and adding them to the cluster. Administrators can refer to existing guidance on how to obtain
+this information.
+
+The API specs will also include a `yumRepositories` field, which allows for a list of `ConfigMaps`
+containing `.repo` configuration files that allow an entitlement to be downloaded from a Red Hat
+Satellite instance.
+
+#### Subscription Bundle Controller
+
+When a `ClusterBundle` object is created, the Subscription Bundle Controller (a new controller)
+will perform the necessary actions to make the bundle available to cluster workloads. This consists
+of the following:
+
+1. Verify that the referenced `entitlements` and optional `yumRepositories` resources exist.
+2. Create the ProjectedResource CSI driver instances for the `entitlements` and (optional)
```

Comment:
what's the behavior if a ClusterBundle is later updated to point to a different entitlement secret?

what happens to the existing ProjectedResources? (also didn't we rename them Shares?)  And what happens to the pods that are already consuming those ProjectedResources?

Reply:
@gabemontero you may be able to weigh in on what the CSI driver would do. IMO the `ClusterBundle` should wire the change through to the respective `Share` object. That in turn takes care of updating the pods/containers mounting in the shared resources.

Reply:
I agree @adambkaplan 

And we update / delete as needed each pod's tmpfs copy of a share's referenced secrets/configmaps as needed 

Ultimately, that will occur if either the secret/configmap or share changes.